2010-06-28  Michael Tautschnig  <mt@debian.org>

	* ainsl: Improved error handling in case target device is full (thanks Rob
		<rclemley@booksys.com>)
	* ainsl.1: Document ainsl exit codes (thanks Rob <rclemley@booksys.com>)
Index: trunk/man/ainsl.1
===================================================================
--- trunk.orig/man/ainsl.1
+++ trunk/man/ainsl.1
@@ -28,6 +28,9 @@
 are used for matching, not when adding the line. Additionaly,
 the following characters are escaped in LINE: ( ) +
 
+The exit code returned by ainsl is 0 on success and non-zero
+on error.  See the section EXIT CODES.
+
 .SH OPTIONS
 .TP
 .B \-a
@@ -64,6 +67,22 @@
 
 Add exports entry without exactly matching spaces.
 
+.SH EXIT CODES
+.TP
+.B 0
+Success: Either FILE contains the LINE/PATTERN or LINE was appended to FILE.
+.TP
+.B 13
+Permissions do not allow writing to FILE and LINE/PATTERN was not found in file.
+.TP
+.B 28
+FILE cannot written because the filesystem has no space and LINE/PATTERN was not
+found in file.
+.TP
+.B 30
+FILE cannot written because the filesystem is read-only and LINE/PATTERN was not
+found in file.
+
 .SH NOTES
 This is a similar function to AppendIfNoSuchLine from cfengine(8).
 .SH SEE ALSO
Index: trunk/bin/ainsl
===================================================================
--- trunk.orig/bin/ainsl
+++ trunk/bin/ainsl
@@ -136,5 +136,5 @@
 print "ainsl: appending to $filename: $line\n" if $verbose;
 exit 0 if $opt_n;
 open (INFILE, ">>$filename") or die "ainsl: can't open $filename for writing. $!";
-print INFILE $line,"\n";
-close(INFILE);
+print INFILE $line,"\n" || die "ainsl: cannot print to file $filename: $!";
+close(INFILE) or die "ainsl: error saving file $filename: $!";
